WooCommerce/WordPressAuthenticator/Unified Auth/View Related/Login/MagicLinkRequestViewController.swift

```diff
@@ -0,0 +1,230 @@
+
+    init(fallbackAction: MagicLinkFallbackAction) {
+        self.fallbackAction = fallbackAction
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override var sourceTag: WordPressSupportSourceTag {
+        get {
+            return .loginMagicLink
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let email = loginFields.username
+        if !email.isValidEmail() {
+            assert(email.isValidEmail(), "The value of loginFields.username was not a valid email address.")
+        }
+
+        tracker.set(flow: .loginWithMagicLink)
+        tracker.track(step: .start)
+
+        configureStackView()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        WordPressAuthenticator.track(.loginMagicLinkRequestFormViewed)
+    }
+}
+
+// MARK: Actions
+private extension MagicLinkRequestViewController {
+    func sendMagicLink() {
+        Task { @MainActor in
+            tracker.track(click: .requestMagicLink)
+            configureSubmitButton(animating: true)
+
+            let result = await MagicLinkRequester().requestMagicLink(email: loginFields.username, jetpackLogin: loginFields.meta.jetpackLogin)
+
+            configureSubmitButton(animating: false)
+
+            switch result {
+            case .success:
+                let vc = MagicLinkRequestedViewController(email: loginFields.username,
+                                                          fallbackAction: fallbackAction) { [weak self] in
+                    self?.openFallbackScreen()
+                }
+
+                vc.loginFields = loginFields
+                navigationController?.pushViewController(vc, animated: true)
+
+            case .failure(let error):
+                WordPressAuthenticator.track(.loginMagicLinkFailed, error: error)
+                displayErrorAlert(Localization.magicLinkError, sourceTag: self.sourceTag)
+            }
+        }
+    }
+
+    func openFallbackScreen() {
+        let vc: LoginViewController?
+        switch self.fallbackAction {
+        case .password:
+            tracker.track(click: .loginWithAccountPassword)
+            vc = PasswordViewController.instantiate(from: .password)
+        case .wpcomUsernamePassword:
+            tracker.track(click: .loginWithWPComUsernamePassword)
+            vc = SiteCredentialsViewController.instantiate(from: .siteAddress)
+        }
+
+        guard let vc else { return }
+
+        vc.loginFields = self.loginFields
+        if fallbackAction == .wpcomUsernamePassword {
+            vc.loginFields.siteAddress = "https://wordpress.com"
+        }
+
+        self.navigationController?.pushViewController(vc, animated: true)
+    }
+}
+
+// MARK: UI Setup
+private extension MagicLinkRequestViewController {
+    private func configureStackView() {
```

Comment:
I didn't use a xib file, as I still don't feel familiar with XCode UI for them, and I didn't use SwiftUI just to make sure the content is correctly styled to match the other screens. If you have any remarks about this please share them.